New num_calls_arbitrary functionality, added testing, cleaned up

jvsr1 opened the pull request:
- Added num_calls_arbitrary functionality: determine how many times a function (passed in arbitrarily!) was called. This was only going to be done for open, but it was easy to add it for arbitrary functions. 
- Also added functionality to extract all comments separately from a python file.
- Added testing, and documentation.

tskluzac reviewed (changes requested):
Great adds @jvsr1 ! What is the "Alternate Match" comment? Is this something to be integrated or removed?

jvsr1 commented:
Hi @tskluzac! thanks for the feedback :) that is simply a different way I found of doing the same thing when I was experimenting with regex -- I've removed it and recommitted.

Update: also added the compilation version extraction and updated the readme file.
